src/main/java/com/marklogic/client/ext/datamovement/listener/ProgressQueryBatchListener.java

```diff
@@ -0,0 +1,154 @@
+package com.marklogic.client.ext.datamovement.listener;
+
+import com.marklogic.client.datamovement.QueryBatch;
+import com.marklogic.client.datamovement.QueryBatchListener;
+import com.marklogic.client.datamovement.QueryBatcher;
+
+import java.math.BigDecimal;
+import java.math.MathContext;
+
+/**
+ * QueryBatchListener for receiving progress updates as QueryBatch instances are processed. Progress updates are
+ * communicated via instances of the nested static ProgressUpdate class, which are given to an implementation of the
+ * nested static ProgressUpdateListener callback interface.
+ */
+public class ProgressQueryBatchListener implements QueryBatchListener {
+
+	private ProgressUpdateListener progressListener;
+
+	private long startTime;
+	private long totalResults;
+	private long resultsCount;
+
+	/**
+	 * Use this constructor for when the total number of results isn't known ahead of time.
+	 *
+	 * @param progressListener
+	 */
+	public ProgressQueryBatchListener(ProgressUpdateListener progressListener) {
```

Comment:
Will there be a case where we have to update the progress to multiple things? If yes, instead of passing it in the constructor, can we have a list of ProgressUpdateListener registered via a method like onProgressUpdate(ProgressUpdateListener)? Similar to [onDocumentReady](https://github.com/marklogic/java-client-api/blob/develop/marklogic-client-api/src/main/java/com/marklogic/client/datamovement/ExportListener.java#L244) of ExportListener.

Reply:
I went back and forth on this originally. My thought was that a client can always achieve that via a composite implementation of ProgressUpdateListener (PUL) that allows for registering multiple PULs. Not supporting a list always means this class doesn't have to worry about what to do if e.g. there are 2 and the first one throws an exception - the client can worry about that instead in a composite implementation.

Reply:
Makes sense. In the past with other out of the box listeners, we have simply logged an error if a PUL threw an error and pretty much done nothing with it. We expect the PUL to handle any exception that it might throw. Thanks!

Reply:
And if you are raising this PR against marklogic-client-api, can we have a list of Java Consumers similar to what we have in ExportListener so that we can be consistent with the rest of the out of the box listeners. We just log the error if any of the consumers throw an exception.